app/models/report.rb

```diff
@@ -15,14 +16,55 @@ def self.generate_report_for_keywords(search_keywords)
     )
   end
 
+  def self.find_for_guid(guid)
+    where(guid: guid).first
+  end
+
+  def find_occupation(occupation_id)
+    occupations.find { |o| o.id == occupation_id }
```

Comment:
is we use `occupations.where(id: occupation_id).first` or `occupations.find(occupation_id)` then ActiveRecord will handle querying the DB or running this in memory for us. As this method stands it will always query the DB for all records, then enumerate over each record in memory.